lib/mix_unused/filter.ex

```diff
@@ -24,23 +24,41 @@ defmodule MixUnused.Filter do
   @spec reject_matching(exports :: Exports.t(), patterns :: [pattern()]) ::
           Exports.t()
   def reject_matching(exports, patterns) do
-    filters =
-      Enum.map(patterns, fn
-        {_m, _f, _a} = entry -> entry
-        {m, f} -> {m, f, :_}
-        {m} -> {m, :_, :_}
-        m when is_atom(m) -> {m, :_, :_}
-        %Regex{} = m -> {m, :_, :_}
-        cb when is_function(cb) -> cb
-      end)
+    filters = normalize_filter_patterns(patterns)
 
     Enum.reject(exports, fn {func, meta} ->
       Enum.any?(filters, &mfa_match?(&1, func, meta))
     end)
     |> Map.new()
   end
 
-  @spec mfa_match?(mfa(), pattern(), Meta.t()) :: boolean()
+  @doc """
+  Get values in `exports` that match any pattern in `patterns`
+  """
+  @spec filter_matching(exports :: Exports.t(), patterns :: [pattern()]) ::
+          Exports.t()
+  def filter_matching(exports, patterns) do
+    filters = normalize_filter_patterns(patterns)
+
+    Enum.filter(exports, fn {func, meta} ->
+      Enum.any?(filters, &mfa_match?(&1, func, meta))
+    end)
+    |> Map.new()
+  end
+
+  @spec normalize_filter_patterns(patterns :: [pattern()]) :: [pattern()]
+  defp normalize_filter_patterns(patterns) do
+    Enum.map(patterns, fn
+      {_m, _f, _a} = entry -> entry
+      {m, f} -> {m, f, :_}
+      {m} -> {m, :_, :_}
+      m when is_atom(m) -> {m, :_, :_}
+      %Regex{} = m -> {m, :_, :_}
+      cb when is_function(cb) -> cb
+    end)
+  end
```

Comment:
Just a refactor opportunity. 🙂 

Instead of `normalize_filter_patterns(patterns :: [pattern()]) :: [pattern()]` you can create a more reusable functional component, for instance:
```
@spec matcher(patterns :: [pattern()]) :: (mfa() -> boolean)
```

Then you can use it in this way:
```
matcher = matcher(patterns)
Map.filter(exports, fn {mfa, _} -> matcher.(mfa) end)
```

If it makes sense, the matcher could also take the whole pair, so we have `Map.filter(exports, matcher(patterns))`.

Reply:
makes sense :rocket: